io/io/src/RRawFile.cxx

```diff
@@ -68,6 +90,23 @@ ROOT::Internal::RRawFile::Create(std::string_view url, ROptions options)
 #ifdef _WIN32
       return std::unique_ptr<RRawFile>(new RRawFileWin(url, options));
 #else
+      // If URL is a file on an EOS FUSE mount, attempt redirection to XRootD protocol.
+      if (gEnv->GetValue("TFile.CrossProtocolRedirects", 1) == 1) {
+         // At the moment the url parameter can only be the file name
+         ssize_t len = getxattr(url.data(), "eos.url.xroot", nullptr, 0);
```

Comment:
I think we should combine the logic here and in TFile into a common header (in RIO) that gives us one function to read an extended attribute.